Library/Homebrew/utils/fork.rb

```diff
@@ -15,7 +15,18 @@ def self.safe_fork(&_block)
             write.fcntl(Fcntl::F_SETFD, Fcntl::FD_CLOEXEC)
             yield
           rescue Exception => e # rubocop:disable Lint/RescueException
-            Marshal.dump(e, write)
+            error_hash = JSON.parse e.to_json
+
+            # Special case: We need to recreate ErrorDuringExecutions
+            # for proper error messages and because other code expects
+            # to rescue them further down.
+            if e.is_a?(ErrorDuringExecution)
```

Comment:
I'm a little stuck on the ducktyping here -- I don't think we want to test the exception with `responds_to?` here, since there are probably other exceptions that have `cmd`/`status`/`output` attributes that we don't want to handle specially. I flattened the exception recreation below, though.